matchzoo/preprocessor/segment.py

```diff
@@ -31,9 +31,11 @@ def segment(inputs: list, stage: str) -> datapack.DataPack:
 
     left = inputs[['id_left', 'text_left']].drop_duplicates(['id_left'])
     left.set_index('id_left', inplace=True)
+    left['length_left'] = left.apply(lambda r: len(r['text_left']), axis=1)
 
     right = inputs[['id_right', 'text_right']].drop_duplicates(['id_right'])
     right.set_index('id_right', inplace=True)
+    right['length_right'] = right.apply(lambda r: len(r['text_right']), axis=1)
```

Comment:
I guess you do not need to specify axis ? see [here](https://stackoverflow.com/questions/34962104/pandas-how-can-i-use-the-apply-function-for-a-single-column)

Reply:
It does not work to remove the `axis`. In fact, the default value of the `axis` is 0. The case you show is ok because it only modified existing values, but I have appended a new column.